authenticator/types.go

```diff
@@ -35,9 +35,27 @@ type TokenResponse struct {
 // when making a call to the Authenticate method
 // It returns the status of the response, headers and the response body
 type AuthenticationResponse struct {
-	Status  int         `json:"status"`
-	Headers http.Header `json:"headers"`
-	Body    interface{} `json:"body,omitempty"`
+	Status  int
+	Headers http.Header
+	body    interface{}
+}
+
+// Error returns the ErrorBody of the authentication response
+func (a *AuthenticationResponse) Error() *ErrorBody {
+	if a.Status != http.StatusOK {
+		return a.body.(*ErrorBody)
+	}
+
+	return nil
+}
+
+// TokenResponse returns the token returned by the response
+func (a *AuthenticationResponse) TokenResponse() *TokenResponse {
+	if a.Status != http.StatusOK {
+		return nil
+	}
+
+	return a.body.(*TokenResponse)
```

Comment:
Should we prefer the panic, which makes it obvious if we've messed up, or do the type assert with the `ok bool`, and then return nil if not ok.  Might make it harder to determine what went wrong, but nicer for the user if it doesn't just panic.

Reply:
Don't think we should panic. I'll go for the latter option.